lib/elixir/lib/task/supervisor.ex

```diff
@@ -67,6 +67,7 @@ defmodule Task.Supervisor do
   def async(supervisor, module, fun, args) do
     args = [self, :link, get_info(self), {module, fun, args}]
     {:ok, pid} = Supervisor.start_child(supervisor, args)
+    Process.link(pid)
```

Comment:
Being asynchronous is nice because it means the Supervisor does not get blocked if the `Process.link` call in the task is attempting to connect to another node (as in the race condition described in the comments in the Task.Supervised module).